docs/streams/developer-guide.html

```diff
@@ -267,21 +295,532 @@
+           <td>
+               Groups the records by the existing key.<p>
+
+               Grouping is a prerequisite for aggregating a stream or a table and ensures that data is properly
+               partitioned ("keyed") for subsequent operations.<p>
+
+               <b>When to set explicit serdes</b>: Variants of <code>groupByKey</code> exist to override the configured default serdes of
+               your application, which you must do if the key and/or value types of the resulting <code>KGroupedStream</code> do
+               not match the configured default serdes.<p>
+
+               <b>Note:</b>
+               Grouping vs. Windowing: A related operation is windowing, which lets you control how to "sub-group" the
+               grouped records of the same key into so-called windows for stateful operations such as windowed aggregations
+               or windowed joins.<p>
+
+               Causes data re-partitioning if and only if the stream was marked for re-partitioning. <code>groupByKey</code> is
+               preferable to <code>groupBy</code> because it re-partitions data only if the stream was already marked for re-partitioning.
+	       However, <code>groupByKey</code> does not allow you to modify the key or key type like <code>groupBy</code> does.
+
+               <pre class="brush: java;">
+               KStream&lt;byte[], String&gt; stream = ...;
+
+               // Group by the existing key, using the application's configured
+               // default serdes for keys and values.
+               KGroupedStream&lt;byte[], String&gt; groupedStream = stream.groupByKey();
+
+               // When the key and/or value types do not match the configured
+               // default serdes, we must explicitly specify serdes.
+               KGroupedStream&lt;byte[], String&gt; groupedStream = stream.groupByKey(
+                   Serdes.ByteArray(), /* key */
+                   Serdes.String()     /* value */
+               );
+               </pre>
+           </td>
+       </tr>
+       <tr>
+           <td><b>GroupBy</b>: <code>KStream->KGroupedStream or KTable->KGroupedTable</code></td>
+           <td>
+               Groups the records by a new key, which may be of a different key type. When grouping a table,
+               you may also specify a new value and value type. groupBy is a shorthand for selectKey(...).groupByKey().<p>
+
+               Grouping is a prerequisite for aggregating a stream or a table and ensures that data is properly
+               partitioned ("keyed") for subsequent operations.<p>
+
+               <b>When to set explicit serdes</b>: Variants of groupBy exist to override the configured default serdes of your
+               application, which you must do if the key and/or value types of the resulting KGroupedStream or
+               KGroupedTable do not match the configured default serdes.
+
+               <b>Note:</b>
```

Comment:
This paragraph should be wrapped into `<p>` or similar, or the previous graph needs to be wrapped, or whatever.  Right now, they are rendered as one single paragraph.

Also, "Note:" should be changed to `<b>Note on Grouping vs. Windowing:</b>`, see above.

Lastly, the quotes around "sub-group" are rendered oddly in my browser.

Screenshot:
<img width="840" alt="screen shot 2017-07-17 at 17 38 11" src="https://user-images.githubusercontent.com/294849/28276274-b7808eec-6b16-11e7-87cf-86e8a53363bb.png">